public/app/features/search/components/SearchWrapper.tsx

```diff
@@ -28,7 +54,17 @@ export const SearchWrapper: FC = () => {
       appEvents.off(CoreEvents.showDashSearch, openSearch);
       appEvents.off(CoreEvents.hideDashSearch, closeOnItemClick);
     };
-  }, [isOpen]);
+  }, [search]);
 
-  return isOpen ? <DashboardSearch onCloseSearch={() => setIsOpen(false)} payload={payload} /> : null;
+  return isOpen ? <DashboardSearch onCloseSearch={closeSearch} payload={payload} /> : null;
+});
+
+const mapStateToProps = (state: StoreState) => {
+  return { search: getLocationQuery(state.location).search };
+};
+
+const mapDispatchToProps = {
+  updateLocation,
 };
```

Comment:
Type both of these using the redux types MapStateToProps MapDispatchToProps  
https://github.com/grafana/grafana/blob/master/public/app/features/dashboard/dashgrid/DashboardPanel.tsx#L143